cpp/cmake/IceConfig.cmake

```diff
@@ -10,69 +11,173 @@ if (NOT EXISTS ${Ice_HOME})
   message(FATAL_ERROR "The specified Ice_HOME directory does not exist: ${Ice_HOME}")
 endif()
 
-find_program(Ice_SLICE2CPP_EXECUTABLE slice2cpp HINTS ${Ice_HOME}/cpp/bin PATH_SUFFIXES x64/Release x64/Debug)
+# List of all available Ice components
+set(Ice_AVAILABLE_COMPONENTS DataStorm Glacier2 Ice IceBox IceGrid IceStorm)
 
-if(NOT DEFINED Ice_SLICE_DIR AND EXISTS ${Ice_HOME}/slice)
-  set(Ice_SLICE_DIR ${Ice_HOME}/slice CACHE PATH "Path to the Ice Slice files directory")
+# IceBT is only available on Linux
+# TODO: Check for required dependency
+if(UNIX AND NOT APPLE)
```

Comment:
We don't always build IceBT on Linux, would be good if we can check if the file is present.